### PR TITLE
RavenDB-16357 Support read-only Certificates in Studio - Part 5

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

**Review by ml054 — CHANGES_REQUESTED** [review]
I've talked about changes with @Danielle9897

**Comment by ml054** [comment]
I need to do little refactor. Will push to this branch.

**Comment by ml054** [comment]
@Danielle9897 Please review my changes. I hope I didn't break anything. 

Things I added:
- ability to use requiredAccess on view w/o active database
- hide current access level in tooltip when required access is cluster admin or operator but user has only database access 
- using accessLevel in functions instead of databaseAccessLevel (and function can now handle all cases)
- added function to infer access type `static isServerWideAccessLevel(access: accessLevel): access is accessLevel { ... }`
- using pure function whereever possible (to avoid illusion that function is pure but uses side effects)
- fail fasts paths in can handle / disable reason tooltips / canAccessView

File: src/Raven.Studio/typescript/common/shell/accessManager.ts [hunk 1]
```diff
@@ -110,51 +108,69 @@ class accessManager {
         const accessLevel = this.activeDatabaseEffectiveAccessLevel();
         return accessLevel === "DatabaseAdmin";
     });
+    
+    static isServerWideAccessLevel(access: accessLevel): access is accessLevel {
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/common/shell/accessManager.ts]: Rename method to Not use the serve-wide concept ?

File: src/Raven.Studio/typescript/common/shell/accessManager.ts [hunk 2]
```diff
@@ -110,51 +108,69 @@ class accessManager {
...
+            return clearance === "ValidUser" ? null : clearance;
+        }
+        
+        return accessManager.databasesAccess[dbName];
+    }
 
-    canHandleOperation(requiredAccess: accessLevel, effectiveDatabaseAccess: databaseAccessLevel): boolean {
-        if (requiredAccess === "DatabaseRead") {
-            return true;
+    canHandleOperation(requiredAccess: accessLevel, actualAccessLevel: accessLevel): boolean {
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/common/shell/accessManager.ts]: static ?

File: src/Raven.Studio/typescript/common/shell/accessManager.ts [hunk 3]
```diff
@@ -110,51 +108,69 @@ class accessManager {
...
+            default: 
+                return false;
         }
     }
     
-    getDisableReasonHtml(dbName: string, requiredAccess: accessLevel, effectiveDbAccess: databaseAccessLevel) {
-        const title = dbName ? "Insufficient database access" :
-                               "Insufficient security clearance";
-        
+    getDisableReasonHtml(requiredAccess: accessLevel, actualAccessLevel: accessLevel) {
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/common/shell/accessManager.ts]: static ?

File: src/Raven.Studio/typescript/common/shell/accessManager.ts [hunk 4]
```diff
@@ -110,51 +108,69 @@ class accessManager {
...
                     <ul>
                         <li>Required: <strong>${requiredText}</strong></li>
                         <li>Actual: <strong>${actualText}</strong></li>
                     </ul>
                 </div>`;
+        } else {
+            return `<div class="text-left">
+                    <h4>Insufficient access</h4>
+                    <ul>
+                        <li>Required: <strong>${requiredText}</strong></li>
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/common/shell/accessManager.ts:174]: IMHO - I prefer the previous tooltips version
This is Not informative enough - can we discuss that ?

File: src/Raven.Studio/typescript/viewmodels/manage/certificates.ts [hunk 5]
```diff
@@ -229,11 +228,8 @@ class certificates extends viewModelBase {
         return accessManager.default.getAccessColor(accessLevel);
     }
 
-    getAccessInfoText(dbName: string, accessLevel: databaseAccessLevel) {
-        return `<div class="text-left padding padding-xs">
-                   <strong>Name</strong>: ${dbName === "All" ? "All databases" : generalUtils.escapeHtml(dbName) }<br/>
-                   <strong>Access</strong>: ${accessManager.default.getAccessLevelText(accessLevel) }
-                </div>`
+    getAccessInfoText(accessLevel: databaseAccessLevel) {
+        return accessManager.default.getAccessLevelText(accessLevel);
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/viewmodels/manage/certificates.ts]: Did you remove the tooltip content on purpose ?

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 1 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts]: done

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 4 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts:174]: done

File: src/Raven.Studio/typescript/viewmodels/manage/certificates.ts (hunk 5 above)
Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/manage/certificates.ts]: done

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 2 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts]: ok

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 4 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts:174]: ok

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 1 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts]: done

File: src/Raven.Studio/typescript/common/shell/accessManager.ts (hunk 3 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/shell/accessManager.ts]: ok

**Review by ml054 — APPROVED** [review]

**Pull request merged**